### PR TITLE
[Enhancement] Donut chart view

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description
Added new cheese-chart-view for showing percentage of each crypto currency which user has in his wallet.

### Issue resolved
Fixes https://github.com/knocte/gwallet/issues/2

### CODE REVIEW

File: src/GWallet.Frontend.XF.Gtk/GWallet.Frontend.XF.Gtk.fsproj [hunk 1]
```diff
@@ -234,6 +240,9 @@
   <!-- workaround for https://github.com/dotnet/corefx/issues/19914 / https://bugzilla.xamarin.com/show_bug.cgi?id=60315 -->
   <Target Name="AfterBuild">
     <Delete Files="bin\$(Configuration)\System.Net.Http.dll" />
+	<Copy SourceFiles="..\..\packages\SkiaSharp.NativeAssets.Linux.1.68.0\runtimes\linux-x64\native\libSkiaSharp.so" 
+		DestinationFiles="bin\$(Configuration)\libSkiaSharp.so" />
```
Comment (knocte) [inline src/GWallet.Frontend.XF.Gtk/GWallet.Frontend.XF.Gtk.fsproj]: cosmetic-style: please use spaces, not tabs

File: src/GWallet.Frontend.XF.Gtk/GWallet.Frontend.XF.Gtk.fsproj [hunk 2]
```diff
@@ -234,6 +240,9 @@
   <!-- workaround for https://github.com/dotnet/corefx/issues/19914 / https://bugzilla.xamarin.com/show_bug.cgi?id=60315 -->
   <Target Name="AfterBuild">
     <Delete Files="bin\$(Configuration)\System.Net.Http.dll" />
+	<Copy SourceFiles="..\..\packages\SkiaSharp.NativeAssets.Linux.1.68.0\runtimes\linux-x64\native\libSkiaSharp.so"
```
Comment (knocte) [inline src/GWallet.Frontend.XF.Gtk/GWallet.Frontend.XF.Gtk.fsproj]: can you add an XML comment pointing to the SkiaSharp bug I opened? Like this:

```
<!-- workaround for https://github.com/mono/SkiaSharp/issues/794 -->
```

**Comment by AndreiMisiukevich** [comment]
@knocte re-review please

**Comment by knocte** [comment]
Can you test the Mac frontend? I'm getting a crash:

```
System.IO.FileNotFoundException: Could not load file or assembly 'SkiaSharp.Extended.Svg' or one of its dependencies
  at System.AppDomain.Load (System.Reflection.AssemblyName assemblyRef, System.Security.Policy.Evidence assemblySecurity) [0x0006f] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/mcs/class/corlib/System/AppDomain.cs:746
  at System.AppDomain.Load (System.Reflection.AssemblyName assemblyRef) [0x00000] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/mcs/class/corlib/System/AppDomain.cs:714
  at at (wrapper remoting-invoke-with-check) System.AppDomain.Load(System.Reflection.AssemblyName)
  at System.Reflection.Assembly.Load (System.Reflection.AssemblyName assemblyRef) [0x00000] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/mcs/class/corlib/System.Reflection/Assembly.cs:600
  at ObjCRuntime.Runtime.CollectReferencedAssemblies (System.Collections.Generic.List`1[T] assemblies, System.Reflection.Assembly assembly) [0x00085] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/ObjCRuntime/Runtime.cs:536
  at ObjCRuntime.Runtime.CollectReferencedAssemblies (System.Collections.Generic.List`1[T] assemblies, System.Reflection.Assembly assembly) [0x00085] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/ObjCRuntime/Runtime.cs:536
  at ObjCRuntime.Runtime.RegisterEntryAssembly (System.Reflection.Assembly entry_assembly) [0x00032] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/ObjCRuntime/Runtime.cs:497
  at ObjCRuntime.Runtime.RegisterAssemblies () [0x00001] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/ObjCRuntime/Runtime.cs:473
  at AppKit.NSApplication.Init () [0x00022] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.2.1.15/src/Xamarin.Mac/AppKit/NSApplication.cs:60
  at GWallet.Frontend.XF.Mac.main.main (System.String[] args) [0x00000] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Frontend.XF.Mac/Main.fs:8
```

**Comment by knocte** [comment]
Shit! Even iOS crashes now, I don't understand:

```
2019-02-23 18:31:32.635535+0800 GWallet.Frontend.XF.iOS[13844:352289] Unhandled managed exception:
@rpath/libSkiaSharp.framework/libSkiaSharp (System.DllNotFoundException)
  at (wrapper managed-to-native) SkiaSharp.SkiaApi.sk_picture_recorder_new()
  at SkiaSharp.SKPictureRecorder..ctor () [0x00000] in <d8e6e9a293b648949e3c95c25ca807ca>:0 
  at SkiaSharp.Extended.Svg.SKSvg.Load (System.Xml.Linq.XDocument xdoc) [0x00266] in <a256cb0a58d44111bf57f22f7287a4c8>:0 
  at SkiaSharp.Extended.Svg.SKSvg.Load (System.Xml.XmlReader reader) [0x00007] in <a256cb0a58d44111bf57f22f7287a4c8>:0 
  at SkiaSharp.Extended.Svg.SKSvg.Load (System.IO.Stream stream) [0x00012] in <a256cb0a58d44111bf57f22f7287a4c8>:0 
  at GWallet.Frontend.XF.Controls.PieChartView.Draw () [0x00201] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Frontend.XF/Controls/PieChartView.fs:196 
  at GWallet.Frontend.XF.Controls.PieChartView.OnPropertyChanged (System.String propertyName) [0x00116] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Frontend.XF/Controls/PieChartView.fs:226 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00116] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:634 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x0015b] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:426 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0003d] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:583 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindablePropertyKey propertyKey, System.Object value) [0x0000e] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:107 
  at Xamarin.Forms.VisualElement.set_Height (System.Double value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:194 
  at Xamarin.Forms.VisualElement.SetSize (System.Double width, System.Double height) [0x0001a] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:907 
  at Xamarin.Forms.VisualElement.set_Bounds (Xamarin.Forms.Rectangle value) [0x0005d] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:186 
  at Xamarin.Forms.VisualElement.Layout (Xamarin.Forms.Rectangle bounds) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:565 
  at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.View child, Xamarin.Forms.Rectangle region, Xamarin.Forms.SizeRequest childSizeRequest) [0x0021a] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:318 
  at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00092] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:66 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x0014b] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:263 
  at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:223 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:682 
  at Xamarin.Forms.VisualElement.SetSize (System.Double width, System.Double height) [0x00021] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:909 
  at Xamarin.Forms.VisualElement.set_Bounds (Xamarin.Forms.Rectangle value) [0x0005d] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:186 
  at Xamarin.Forms.VisualElement.Layout (Xamarin.Forms.Rectangle bounds) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:565 
  at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.View child, Xamarin.Forms.Rectangle region, Xamarin.Forms.SizeRequest childSizeRequest) [0x0021a] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:318 
  at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00092] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:66 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x0014b] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:263 
  at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:223 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:682 
  at Xamarin.Forms.VisualElement.SetSize (System.Double width, System.Double height) [0x00021] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:909 
  at Xamarin.Forms.VisualElement.set_Bounds (Xamarin.Forms.Rectangle value) [0x0005d] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:186 
  at Xamarin.Forms.VisualElement.Layout (Xamarin.Forms.Rectangle bounds) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:565 
  at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.VisualElement child, Xamarin.Forms.Rectangle region) [0x001cf] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:177 
  at Xamarin.Forms.Page.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x0010d] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:187 
  at Xamarin.Forms.Page.UpdateChildrenLayout () [0x000c6] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:259 
  at Xamarin.Forms.Page.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:240 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:682 
  at Xamarin.Forms.VisualElement.SetSize (System.Double width, System.Double height) [0x00021] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:909 
  at Xamarin.Forms.VisualElement.set_Bounds (Xamarin.Forms.Rectangle value) [0x0005d] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:186 
  at Xamarin.Forms.VisualElement.Layout (Xamarin.Forms.Rectangle bounds) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:565 
  at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.VisualElement child, Xamarin.Forms.Rectangle region) [0x00054] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:146 
  at Xamarin.Forms.Page.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x0010d] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:187 
  at Xamarin.Forms.Page.UpdateChildrenLayout () [0x000c6] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:259 
  at Xamarin.Forms.Page.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:240 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:682 
  at Xamarin.Forms.Page.ForceLayout () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:154 
  at Xamarin.Forms.NavigationPage.InsertPageBefore (Xamarin.Forms.Page page, Xamarin.Forms.Page before) [0x000d1] in D:\a\1\s\Xamarin.Forms.Core\NavigationPage.cs:325 
  at Xamarin.Forms.NavigationPage+NavigationImpl.OnInsertPageBefore (Xamarin.Forms.Page page, Xamarin.Forms.Page before) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\NavigationPage.cs:437 
  at Xamarin.Forms.Internals.NavigationProxy.InsertPageBefore (Xamarin.Forms.Page page, Xamarin.Forms.Page before) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\NavigationProxy.cs:59 
  at Xamarin.Forms.Internals.NavigationProxy.OnInsertPageBefore (Xamarin.Forms.Page page, Xamarin.Forms.Page before) [0x0003e] in D:\a\1\s\Xamarin.Forms.Core\NavigationProxy.cs:155 
  at Xamarin.Forms.Internals.NavigationProxy.InsertPageBefore (Xamarin.Forms.Page page, Xamarin.Forms.Page before) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\NavigationProxy.cs:59 
  at GWallet.Frontend.XF.FrontendHelpers+SwitchToNewPageDiscardingCurrentOne@207.Invoke () [0x00040] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Frontend.XF/FrontendHelpers.fs:215 
  at Foundation.NSAsyncActionDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.15/src/Xamarin.iOS/Foundation/NSAction.cs:152 
--- End of stack trace from previous location where exception was thrown ---
  at GWallet.Backend.Infrastructure.Report[a] (System.Exception ex) [0x0000b] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Backend/Infrastructure.fs:22 
  at GWallet.Backend.Infrastructure.OnUnhandledException[a] (System.Object sender, System.UnhandledExceptionEventArgs args) [0x00000] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Backend/Infrastructure.fs:28 
  at GWallet.Backend.Infrastructure+SetupSentryHook@31-4.Invoke (System.Object delegateArg0, System.UnhandledExceptionEventArgs delegateArg1) [0x00000] in /Users/andres/Documents/Code/gwalletFRONTENDMERGE/src/GWallet.Backend/Infrastructure.fs:31

```

**Comment by knocte** [comment]
But GTK and Android work!

**Comment by AndreiMisiukevich** [comment]
@knocte yes, i will test it.
Seems it's build issue.. remove bin/obj files usually fixes such cases

**Comment by knocte** [comment]
You were right! After doing `git clean -fdx` and rebuilding, it works.

**Comment by knocte** [comment]
I'll merge this tomorrow morning.

**Comment by AndreiMisiukevich** [comment]
@knocte cool) look forward to my first merged PR :)

**Pull request merged**